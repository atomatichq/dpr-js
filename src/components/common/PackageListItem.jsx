import React, { PropTypes } from 'react'
import { Link } from 'react-router'
import ReactMarkdown from 'react-markdown'
import { Row, Col, Button } from 'react-bootstrap'

class PackageListItem extends React.Component {

  constructor(props) {
    super(props)
  }

  render() {
    const { name, title, description, publisherName } = this.props
    const packageUrl = `${publisherName}/${name}`
    return (
      <div className="card mb-2">
        <div className="card-block">

          <h5 className="card-title"><span className="fa fa-files-o" />{title}</h5>
          <h6>[{name}]</h6>

          { description !== undefined ?
            <ReactMarkdown className="panel-body" source={description} />
            : ''
          }

          <Row className="show-grid clearfix">
            <Col sm={2}>
              <img src="https://avatars3.githubusercontent.com/u/22451011?v=3&s=200" className="img-fluid rounded-circle" />
            </Col>
            <Col sm={6} className="align-self-center">
              <small><Link to={`/${publisherName}`}>by:{publisherName}</Link></small>
            </Col>
            <Col sm={4} className="pull-right align-self-center">
              <Button bsStyle="default" bsSize="xsmall">
                <Link to={packageUrl}>Explore</Link>
              </Button>
            </Col>
          </Row>
        </div>
      </div>
    )
  }
}

PackageListItem.propTypes = {
  name: PropTypes.string.isRequired
  , description: PropTypes.string
  , publisherName: PropTypes.string.isRequired
  , title: PropTypes.string.isRequired
}

export default PackageListItem
