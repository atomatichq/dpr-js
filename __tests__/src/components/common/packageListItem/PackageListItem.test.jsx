import React from 'react'
import renderer from 'react-test-renderer'
import PackageListItem from '../../../../../src/components/common/PackageListItem'

test('Rendered output matches previously created snapshot', () => {
  const component = renderer.create(<PackageListItem
    name="finance-vix"
    description="Awesome package"
    title="Finance Vix Package"
    publisherName="publisher"
  />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
