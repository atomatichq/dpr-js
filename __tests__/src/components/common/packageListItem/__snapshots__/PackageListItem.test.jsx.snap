exports[`test Rendered output matches previously created snapshot 1`] = `
<div
  className="card mb-2">
  <div
    className="card-block">
    <h5
      className="card-title">
      <span
        className="fa fa-files-o" />
      Finance Vix Package
    </h5>
    <h6>
      [
      finance-vix
      ]
    </h6>
    <div
      className="panel-body">
      <p>
        Awesome package
      </p>
    </div>
    <div
      className="show-grid clearfix row">
      <div
        className="col-sm-2">
        <img
          className="img-fluid rounded-circle"
          src="https://avatars3.githubusercontent.com/u/22451011?v=3&s=200" />
      </div>
      <div
        className="align-self-center col-sm-6">
        <small>
          <a
            onClick={[Function]}
            style={Object {}}>
            by:
            publisher
          </a>
        </small>
      </div>
      <div
        className="pull-right align-self-center col-sm-4">
        <button
          className="btn btn-xs btn-default"
          disabled={false}
          type="button">
          <a
            onClick={[Function]}
            style={Object {}}>
            Explore
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
